File: Awesome.AI.Source/Awesome.AI/CoreSystems/Whistle.cs
using Awesome.AI.Core;

namespace Awesome.AI.CoreSystems
{
    public class Whistle
    {
        private TheMind mind;
        public Whistle(TheMind mind)
        {
            this.mind = mind;
            this.mind.result_whistle = "";
        }

        private string[] gimmick = { "[.??]", "[??.]" };
        private int count = 0;
        private int num = 0;
        public void Do(bool _pro)
        {
            if (!_pro)
                return;

            if (count > 1)
                count = 0;

            bool res = mind._quick.Result("WHISTLE");

            if (res)
                num++;

            if (res)
                mind.result_whistle = $"[{num}][Whistling to my self..]";
            
            if (!res)
                mind.result_whistle = $"[{num}]" + gimmick[count];

            count++;
        }
    }
}

File: Awesome.AI.Python/CoreSystems/Whistle.py
"""Whistle output behavior ported from ``Whistle.cs``."""


class Whistle:
    def __init__(self, mind):
        self.mind = mind
        self.mind.result_whistle = ""
        self.gimmick = ("[.??]", "[??.]")
        self.count = 0
        self.num = 0

    def Do(self, _pro):
        if not _pro:
            return

        if self.count > 1:
            self.count = 0

        result = self.mind._quick.Result("WHISTLE")
        if result:
            self.num += 1
            self.mind.result_whistle = f"[{self.num}][Whistling to my self..]"
        else:
            self.mind.result_whistle = f"[{self.num}]" + self.gimmick[self.count]

        self.count += 1
